src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -494,45 +497,86 @@ public static void close(Closeable closeable) {
     /**
      * Resolve the specified filename to a URI.
      *
-     * @param filename name os the file
+     * @param filename name of the file
      * @return resolved header file URI
      * @throws CheckstyleException on failure
      */
     public static URI getUriByFilename(String filename) throws CheckstyleException {
-        // figure out if this is a File or a URL
+        URI uri = getWebOrFilePrefixUri(filename);
+
+        if (uri == null) {
+            uri = getClasspathOrFilepathUri(filename);
+        }
+
+        return uri;
+    }
+
+    /**
+     * Resolves the specified filename containing 'http', 'https', 'ftp',
+     * and 'file' protocols (or any RFC 2396 compliant URL) to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved header file URI or null if URL is malformed or non-existent
+     */
+    public static URI getWebOrFilePrefixUri(String filename) {
         URI uri;
         try {
             final URL url = new URL(filename);
             uri = url.toURI();
         }
-        catch (final URISyntaxException | MalformedURLException ignored) {
+        catch (URISyntaxException | MalformedURLException ignored) {
             uri = null;
         }
+        return uri;
+    }
 
-        if (uri == null) {
+    /**
+     * Resolves the specified local filename to a URI.
+     *
+     * @param filename name of the file
+     * @return resolved header file URI
+     * @throws CheckstyleException on failure
+     */
+    private static URI getClasspathOrFilepathUri(String filename) throws CheckstyleException {
+        final URI uri;
+        if (new File(filename).exists()) {
             final File file = new File(filename);
-            if (file.exists()) {
-                uri = file.toURI();
-            }
-            else {
-                // check to see if the file is in the classpath
-                try {
-                    final URL configUrl;
-                    if (filename.charAt(0) == '/') {
-                        configUrl = CommonUtil.class.getResource(filename);
-                    }
-                    else {
-                        configUrl = ClassLoader.getSystemResource(filename);
-                    }
-                    if (configUrl == null) {
-                        throw new CheckstyleException(UNABLE_TO_FIND_EXCEPTION_PREFIX + filename);
-                    }
-                    uri = configUrl.toURI();
-                }
-                catch (final URISyntaxException ex) {
-                    throw new CheckstyleException(UNABLE_TO_FIND_EXCEPTION_PREFIX + filename, ex);
-                }
-            }
+            uri = file.toURI();
+        }
+        else {
+            uri = getResourceFromClassPath(CLASSPATH_URL_PREFIX_PATTERN
+                .matcher(filename)
+                .replaceAll(Matcher.quoteReplacement("")));
```

Comment:
Not saying what you have is wrong, or has to change but it seems excessive.

We basically just need:
````
if (path starts with classpath)
getResourceFromClassPath(cut off everything after classpath);
````
Wouldn't require a regular expression or anything.

Reply:
Changed to simple String.replace()

Reply:
Team City does not like the simple replace.  IntelliJ recommends using a compiled pattern and matcher.  From https://teamcity.jetbrains.com/viewLog.html?buildId=3291321&buildTypeId=Checkstyle_IdeaInspectionsPullRequest&tab=Inspection:
```
547: getFilepathOrClasspathUri() replace() could be replaced with compiled 'java.util.regex.Pattern' construct 
```

I am going to change this back to the matcher.

Reply:
ok.

Reply:
Reverted back to matcher.